dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ScriptRestRepository.java

```diff
@@ -56,29 +57,24 @@ public class ScriptRestRepository extends DSpaceRestRepository<ScriptRest, Strin
     @Autowired
     private DSpaceRunnableParameterConverter dSpaceRunnableParameterConverter;
 
-    // TODO: findOne() currently requires site ADMIN permissions as all scripts are admin-only at this time.
-    // If scripts ever need to be accessible to Comm/Coll Admins, we would likely need to create a new GrantedAuthority
-    // for Comm/Coll Admins in EPersonRestAuthenticationProvider to use on this endpoint
     @Override
-    @PreAuthorize("hasAuthority('ADMIN')")
+    // authorization is verified inside the method
+    @PreAuthorize("permitAll()")
```

Comment:
This is IMHO an unnecessary constraint. We could have scripts allowed to anonymous user and we actually already have implemented some for DSpace-CRIS. Porting such scripts in DSpace is out-of-scope of this PR but I would like to give to DSpace the flexibility to implement such scripts when needed. The default implementation of the `isAllowedToExecute` method in the abstract class make all the scripts by default restricted to administrator so I don't see a security risk on allowing that. Anyway, if you prefer to restrict the method to authenticated user we can change that

Reply:
Please note that this was also explicitly stated in the REST Contract, see 
https://github.com/DSpace/RestContract/pull/229/files#diff-6672e1b6c27c3e7a23ad35ebaeb95f8b25f0958574fea1409f44cdbf3982df35R180
https://github.com/DSpace/RestContract/pull/229/files#diff-6672e1b6c27c3e7a23ad35ebaeb95f8b25f0958574fea1409f44cdbf3982df35R224

Reply:
@abollini : My concern is that we may not have tests in place which verify this change *for every existing script*.  Previously we chose to make scripts Admin-only for extra security. However, if we are now making them all anonymous-access by default, then we need to verify in each script that doesn't accidentally open up a security hole in an existing script.  That'd require a thorough review of tests for all scripts to verify each is currently blocking anonymous access.

So, I'm not against the approach you suggest. I was suggesting a quicker approach to (at least temporarily) lock these down to requiring authentication.  At this point, I don't have time to re-review every other script and verify this change doesn't accidentally impact one.  

We'd also need to clearly document this level of change... if any other developer has built a custom script and isn't properly checking permissions, their script would potentially be anonymous access.

Reply:
Ok, I'm going to restrict the endpoints to authenticated user and update test and contract accordingly.
BTW, all the scripts have been refactored so that now we have just a single implementation of the isAllowedToExecute method in the abstract class that is inherited by all. This is easily to check using the IDE, this is for instance what my eclipse says
![image](https://github.com/DSpace/DSpace/assets/1537527/16cb1380-193a-4d55-8a11-f9a3315b0719)
only two script have custom implementation of the method. The first has dedicated tests in this PR the second one always return false.

Reply:
done, see this commit https://github.com/DSpace/DSpace/pull/8830/commits/32cd24b7538694ec964289bb13027d09ae06b829